packages/hardhat-core/src/internal/core/tasks/task-definitions.ts

```diff
@@ -36,30 +44,52 @@ export class SimpleTaskDefinition implements TaskDefinition {
   private _positionalParamNames: Set<string>;
   private _hasVariadicParam: boolean;
   private _hasOptionalPositionalParam: boolean;
+  private _scope?: string;
+  private _task: string;
   private _description?: string;
 
   /**
    * Creates an empty task definition.
    *
    * This definition will have no params, and will throw a HH205 if executed.
    *
-   * @param name The task's name.
+   * @param taskIdentifier The task's identifier.
    * @param isSubtask `true` if the task is a subtask, `false` otherwise.
+   * @param _setScopeCallback fn to call while setting the scope.
    */
   constructor(
-    public readonly name: string,
+    taskIdentifier: TaskIdentifier,
+    private readonly _setScopeCallback: (
```

Comment:
shouldn't we default this parameter to an empty function? that way we can avoid having to pass it every time we create a new instance